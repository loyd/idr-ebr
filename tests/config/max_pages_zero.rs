use idr_ebr::{Config, Idr};

struct InvalidConfig;
impl Config for InvalidConfig {
    const MAX_PAGES: u32 = 0;
}

fn main() {
    let _ = Idr::<u64, InvalidConfig>::new();
}
