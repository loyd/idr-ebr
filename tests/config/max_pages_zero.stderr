error[E0080]: evaluation of `<InvalidConfig as idr_ebr::config::ConfigPrivate>::ENSURE_VALID` failed
 --> src/config.rs
  |
  |         assert!(Self::MAX_PAGES > 0);
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'assertion failed: Self::MAX_PAGES > 0', $DIR/src/config.rs:80:9
  |
  = note: this error originates in the macro `assert` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
  --> src/lib.rs
   |
   |         assert!(C::ENSURE_VALID);
   |                 ^^^^^^^^^^^^^^^

note: the above error was encountered while instantiating `fn idr_ebr::Idr::<u64, InvalidConfig>::new`
 --> tests/config/max_pages_zero.rs:9:13
  |
9 |     let _ = Idr::<u64, InvalidConfig>::new();
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
